### PR TITLE
Chained AssertJ assertions should be simplified to the corresponding dedicated assertion

### PR DESCRIPTION
**Pull request by velo**
This is not ready yet, but working on it.

From time to time, I get toasted by a sonar code smell, and I decided to make something about it
https://next.sonarqube.com/sonarqube/coding_rules?open=java%3AS5838&rule_key=java%3AS5838

## What's changed?

For now, created a test for all rules and start implementing missing recipes.

For now, I'm creating a PR to ask questions.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [ ] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/testing/assertj/IsEqualToNull.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+/**
+ * AssertJ has a more idiomatic way of asserting that an object is null. This
+ * recipe will find instances of:
+ * <p>
+ * -`assertThat(Object).isEqualTo(null)` and replace them with `isNull()`.
+ */
+public class IsEqualToNull extends Recipe {
+
+    private static final MethodMatcher IS_EQUAL_TO = new MethodMatcher("org.assertj.core.api.AbstractAssert isEqualTo(..)");
```
Comment (velo) [inline src/main/java/org/openrewrite/java/testing/assertj/IsEqualToNull.java:40]: So, is there any way I can match all subclasses of AbstractAssert?

for instance this recipe is valid for Strings too
```
org.assertj.core.api.AbstractStringAssert isEqualTo(java.lang.String)
```


But I have no idea what all permutations of `isEqualTo` exist out there.

So, is there any way I can catch them all?!

Comment (timtebeek) [reply src/main/java/org/openrewrite/java/testing/assertj/IsEqualToNull.java]: Yes absolutely! The method matcher has an optional second argument boolean to match overrides.
image

I've applied that and a small change to switch to `Traits.methodAccess` already.

**Review by timtebeek — COMMENTED** [review]
Great work here @velo! I like how you went ahead and got a comprehensive view of what is and isn't covered already through those parameterized tests. It looks like we already pass ~2/3 of the tests here.

image

With some adjustments I think we can cover a few more common cases, before perhaps delegating the others to a follow up PR. Let me know your thoughts & plans here! :)

**Comment by velo** [comment]
I will get them sorted

**Comment by timtebeek** [comment]
Hi @velo, are you ok with me polishing this up and merging a smaller increment already? I'm thinking we comment out the failing cases and merge the already nice improvements on this branch.

**Comment by velo** [comment]
Hi @timtebeek 

Sorry, I totally reduced the score of the change to make it a bite size change meant to be merged, but somehow forgot to write an update!

**Comment by velo** [comment]
Hi @timtebeek 

Sorry, I totally reduced the score of the change to make it a bite size change meant to be merged, but somehow forgot to write an update! 

Go for it

**Review by timtebeek — APPROVED** [review]
Nice improvements here! Like how we can cover more cases with less classes; Should help extend the current offerings. I've renamed & revised the classes a bit to make it clear when to use which. Hope you agree & thanks again!

**Pull request merged**